sql/src/main/codegen/includes/common.ftl

```diff
@@ -18,58 +18,51 @@
  */
 
 // Using fully qualified name for Pair class, since Calcite also has a same class name being used in the Parser.jj
-org.apache.druid.java.util.common.Pair<Granularity, String> PartitionGranularity() :
+SqlNode PartitionGranularity() :
 {
   SqlNode e;
-  Granularity granularity;
-  String unparseString;
+  SqlNode result;
 }
 {
   (
     <HOUR>
     {
-      granularity = Granularities.HOUR;
-      unparseString = "HOUR";
+      result = SqlLiteral.createCharString(DruidSqlParserUtils.HOUR_GRAIN, getPos());
```

Comment:
I wonder if it would be possible to use `SqlLiteral.createSymbol` here instead; that could remove the need for the string based matching as well...

Reply:
using `SqlLiteral.createSymbol` as you suggested